contents/swift/welcome-to-swift.md

```diff
@@ -144,9 +146,45 @@ Here are some of the benefits of using `guard` statement over `if-let` statement
 1. Unlike the `if-let` statement, using the `guard` statement causes `past` to remain defined and can be used till the function exits.
 1. While using `if-let` statements can lead to deeply nested `if-let` statements (i.e. pyramid of doom), `guard` statements allow us to have the happy path to be not indented, thereby increasing code readability.
 
-## Data Types
+### Benefit: Handle Post-operations easily
+
+The `defer` key word in Swift provides an easy and safe way to execute some code before leaving current scope. It will be helpful if you need to do post-operations in the function which has many return points.
+
+The following code is an example of using a file:
+```swift
+let fileDescriptor = open(url.path, O_EVTONLY)
+if fileDescriptor == -1 {
+	close(fileDescriptor)
+	return "Failed"
+}
+
+// Use file descriptor
+
+close(fileDescriptor)
+return "Success"
+```
+
+Suppose that you have many failed cases, so you have to close the file for each case, which requires a lot of extra work. However, you can now consolidate all of them to only one `defer` statement:
+
+```swift
+let fileDescriptor = open(url.path, O_EVTONLY)
+
+defer {
+    close(fileDescriptor)
+}
+
+if fileDescriptor == -1 {
+	return "Failed"
+}
+// Use file descriptor
+return "Success"
+```
+
+Through this implementation, the file will be closed no matter which branch the program returns. It also prevents the developer forgetting closing the file in some cases. 
```

Comment:
Using this, the file will be closed no matter which branch the program returns. It also has the added advantage of preventing the developer from forgetting to close the file in some cases.